packages/relay-runtime/util/RelayConcreteNode.js

```diff
@@ -43,30 +43,20 @@ export type ProvidedVariablesType = {+[key: string]: {get(): mixed}};
 
 /**
  * Contains the parameters required for executing a GraphQL request.
- * The operation can either be provided as a persisted `id` or `text`. If given
- * in `text` format, a `cacheID` as a hash of the text should be set to be used
- * for local caching.
+ * The operation can either be provided as a persisted `id` or `text` or both.
+ * If `text` format is provided, a `cacheID` as a hash of the text should be set
+ * to be used for local caching.
  */
-export type RequestParameters =
-  | {
-      +id: string,
-      +text: null,
-      // common fields
-      +name: string,
-      +operationKind: 'mutation' | 'query' | 'subscription',
-      +providedVariables?: ProvidedVariablesType,
-      +metadata: {[key: string]: mixed, ...},
-    }
-  | {
-      +cacheID: string,
-      +id: null,
-      +text: string | null,
-      // common fields
-      +name: string,
-      +operationKind: 'mutation' | 'query' | 'subscription',
-      +providedVariables?: ProvidedVariablesType,
-      +metadata: {[key: string]: mixed, ...},
-    };
+export type RequestParameters = {
+  +cacheID?: string,
+  +id: string | null,
+  +text: string | null,
+  // common fields
```

Comment:
This is needed in the implementation of the `fetch` function, so that you have correct types when sending both values (feature flagged) to the server as you migrate